README.md

```diff
@@ -1,3 +1,55 @@
-# SwornDisk v2
+# MlsDisk
 
-SwornDisk v2 adopts on a new layered design and implementation.
+## Introduction
+
+MlsDisk is a **multilayered log-structured secure virtual disk for TEEs**, which has the following key features:
+
+* **Transparent protection.** As a virtual disk, MlsDisk can _transparently_ protect any file system (e.g., Ext4) that is stacked upon it and runs inside a TEE from a strong adversary outside the Trusted Execution Environment (TEE).
+
+* **Strong security.** MlsDisk promises six security properties: _confidentiality_, _integrity_, _freshness_, _consistency_, _atomicity_, and _irreversibility_. For more information, see [the security goal](#security-goal) below.
+
+* **High performance.** Thanks to its unique log-structured, amplification-minimized design, MlsDisk can deliver an excellent I/O performance that is close to the theoretically optimal level.
+
+## Security Goal
+
+MlsDisk targets a typical setting of TEE usage, where applications are ported into the TEE with no (or few) modifications thanks to a TEE-aware runtime. For enclave TEEs (e.g., Intel SGX), one popular choice for such a runtime is library OSes (e.g., [Occlum](https://github.com/occlum/occlum)). For VM TEEs (e.g., AMD SEV), one can choose off-the-shelf OS kernels like Linux.
+
+![The threat model of MlsDisk.](./docs/figures/mlsdisk-threat-model.png)
+
+As shown in the image above, the TEE runtime is integrated with MlsDisk, which serves as a secure block device that supports three standard block I/O commands including `read()`, `write()`, and `sync()`. From the perspective of MlsDisk's users (e.g., file systems), all data written to or read from MlsDisk is in plaintext. To serve these I/O requests securely, MlsDisk takes some extra security measures, including but not limited to encrypting/decrypting the data transferred to/from the host block device properly.
+
+To distinguish between the addresses on the trusted logical block device (i.e., MlsDisk) and on the untrusted host block device, we term the former as _logical block addresses (LBAs)_ and the latter _host block addresses (HBAs)_.
+
+The security goal of MlsDisk is to provide to its users (e.g., file systems) the following six security guarantees:
```

Comment:
> The security goal of ... is to provide ... security guarantees:

The beginning ("security goal") and the end ("security guarantees") of the sentence are redundant with each other.  Removing one of them.